test/patched/PageWithPreviewServer.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Patch the `server` variable from `page-with/server`, which is a mutable
+ * variable that is updated during execution of `createBrowser` to contain
+ * the connection info of the preview server which was created for the page.
+ *
+ * This function needs to be called from a scope where that mutable variable
+ * is available. That's why we call it in `jest.browser.setup.ts`.
+ *
+ * When the upstream bug is fixed, remove this file and remove the code that
+ * calls it in `jest.browser.setup.ts`
+ *
+ */
+export const patchServerConnectionInfo = (server: PreviewServer) => {
```

Comment:
This isn't really a patch - we just mutate the invalid object (`connectionInfo` with invalid `url`) that is created by `PreviewServer.listen()` when it returns `this.connectionInfo`. So the upstream fix is in the `PreviewServer.listen()` method.

We call this function in `jest.browser.setup.ts`, in order to mutate the `server` object exported from `page-with` and updated via the `createBrowser` promise (subtle, but it works).